scripts/rabbitmq-env

```diff
@@ -369,6 +545,122 @@ run_escript()
         -extra "$escript" "$@"
 }
 
+# TODO: rmq_env_ prefix
+rabbitmq_config()
+{
+    _rmq_env_0="$1"
+
+    # Check that advanced config file has the .config extension
+    # Add .config extension if it's empty
+    RABBITMQ_ADVANCED_CONFIG_FILE_NOEX="$(_rmq_env_get_noex "$RABBITMQ_ADVANCED_CONFIG_FILE")"
+    if [ "$RABBITMQ_ADVANCED_CONFIG_FILE_NOEX.config" = "$RABBITMQ_ADVANCED_CONFIG_FILE" ] || \
+        [ "$RABBITMQ_ADVANCED_CONFIG_FILE_NOEX" = "$RABBITMQ_ADVANCED_CONFIG_FILE" ]; then
+        RABBITMQ_ADVANCED_CONFIG_FILE="$RABBITMQ_ADVANCED_CONFIG_FILE_NOEX.config"
+    fi
+
+    RABBITMQ_CONFIG_FILE_NOEX="$(_rmq_env_get_noex "$RABBITMQ_CONFIG_FILE")"
+
+    # Extension is not specified.
+    # Determine config type from file
+    if [ "$RABBITMQ_CONFIG_FILE_NOEX" = "$RABBITMQ_CONFIG_FILE" ]; then
+        if [ -f "$RABBITMQ_CONFIG_FILE_NOEX.config" ]; then
+            if [ -f "$RABBITMQ_CONFIG_FILE_NOEX.conf" ]; then
+                # Both files exist. Print a warning
+                echo "WARNING: Both old (.config) and new (.conf) format config files exist."
+                echo "WARNING: Using the old format config file: $RABBITMQ_CONFIG_FILE_NOEX.config"
+                echo "WARNING: Please update your config files to the new format and remove the old file"
+            fi
+            RABBITMQ_CONFIG_FILE="$RABBITMQ_CONFIG_FILE_NOEX.config"
+        elif [ -f "$RABBITMQ_CONFIG_FILE_NOEX.conf" ]; then
+            RABBITMQ_CONFIG_FILE="$RABBITMQ_CONFIG_FILE_NOEX.conf"
+        else
+            if [ -f "$RABBITMQ_ADVANCED_CONFIG_FILE" ]; then
+                echo "WARNING: Using RABBITMQ_ADVANCED_CONFIG_FILE: $RABBITMQ_ADVANCED_CONFIG_FILE"
+            fi
+            # No config file exist. Use advanced config for -config arg.
+            RABBITMQ_CONFIG_ARG_FILE="$RABBITMQ_ADVANCED_CONFIG_FILE"
+            RABBITMQ_CONFIG_FILE=""
+        fi
+    fi
+
+    # Set the -config argument.
+    # The -config argument should not have extension.
+    # the file should exist
+    # the file should be a valid erlang term file
+    # Config file extension is .config
+    if [ "$RABBITMQ_CONFIG_FILE_NOEX.config" = "$RABBITMQ_CONFIG_FILE" ]; then
+        RABBITMQ_CONFIG_ARG_FILE="$RABBITMQ_CONFIG_FILE"
+    # Config file extension is .conf
+    elif [ "$RABBITMQ_CONFIG_FILE_NOEX.conf" = "$RABBITMQ_CONFIG_FILE" ]; then
+        RABBITMQ_CONFIG_ARG_FILE="$RABBITMQ_ADVANCED_CONFIG_FILE"
+    elif [ "x" != "x$RABBITMQ_CONFIG_FILE" ] && \
+        [ "$RABBITMQ_CONFIG_FILE_NOEX" != "$RABBITMQ_CONFIG_FILE" ]; then
+        # Config file has an extension, but it's neither .conf or .config
+        echo "ERROR: Wrong extension for RABBITMQ_CONFIG_FILE: $RABBITMQ_CONFIG_FILE" 2>&1
+        echo "ERROR: extension should be either .conf or .config" 2>&1
+        return 1
```

Comment:
This exit should stop the rabbitmq-server sctipt, but it doesn't.
Is there a `set -e` missing anywhere?
I wonder if this check should go to `rabbitmq-server` to keep this file clean from logic.

Reply:
That is surprising. There is a `set -e` in the `_rmq_env_initialize` function but only if `RABBITMQ_SCRIPTS_DIR` is unset. Let me change the code to check the result of `rabbitmq_config`

Reply:
Looks fine after 55a695307f8480f5eaf6e01bd0c6edd0112dbdc4